planemo/workflow_lint.py

```diff
@@ -191,12 +191,12 @@ def _lint_best_practices(path: str, lint_context: WorkflowLintContext) -> None:
     """
 
     def check_json_for_untyped_params(j):
-        values = j if type(j) == list else j.values()
+        values = j if isinstance(j, list) else j.values()
```

Comment:
Can you check this is a dict instance instead ? that should be less narrow. Also the check in line 196 isn't really how you'd do this

Reply:
> Can you check this is a dict instance instead ? 

So `is TYPE` everywhere?

> Also the check in line 196 isn't really how you'd do this

Also with `is`?

Reply:
No, type is type is the most narrow of them all which will not work for any subclasses. `.values()` will only work for objects that satisfy Mapping, while any j that is Iterable will work. So check for the specific instance where you need to call `.values()` first `isinstance(j, Mapping)` and don't worry about the other types that will all be fine.